explainability-service/src/main/java/org/kie/trustyai/service/endpoints/explainers/local/TSSaliencyEndpoint.java

```diff
@@ -0,0 +1,112 @@
+package org.kie.trustyai.service.endpoints.explainers.local;
+
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.microprofile.openapi.annotations.tags.Tag;
+import org.jboss.logging.Logger;
+import org.kie.trustyai.connectors.kserve.v2.CodecParameter;
+import org.kie.trustyai.connectors.kserve.v2.KServeConfig;
+import org.kie.trustyai.connectors.kserve.v2.KServeV2GRPCPredictionProvider;
+import org.kie.trustyai.explainability.local.tssaliency.TSSaliencyExplainer;
+import org.kie.trustyai.explainability.local.tssaliency.TSSaliencyModelWrapper;
+import org.kie.trustyai.explainability.model.*;
+import org.kie.trustyai.service.config.ServiceConfig;
+import org.kie.trustyai.service.endpoints.explainers.ExplainerEndpoint;
+import org.kie.trustyai.service.payloads.requests.explainers.ModelConfig;
+import org.kie.trustyai.service.payloads.requests.explainers.tssaliency.TSSaliencyParameters;
+import org.kie.trustyai.service.payloads.requests.explainers.tssaliency.TSSaliencyRequest;
+
+@Tag(name = "TSSaliency Explainer Endpoint")
+@Path("/explainers/local/tssaliency")
+public class TSSaliencyEndpoint extends ExplainerEndpoint {
+
+    private static final Logger LOG = Logger.getLogger(TSSaliencyEndpoint.class);
+    private static double[] DEFAULT_BASE_VALUE = new double[0];
+    @Inject
+    ServiceConfig serviceConfig;
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response explain(TSSaliencyRequest request) {
+
+        // Instantiate the model from the request
+        final ModelConfig modelConfig = request.getModelConfig();
+        final KServeConfig kServeConfig = KServeConfig.create(modelConfig.getTarget(), modelConfig.getName(), modelConfig.getVersion(), CodecParameter.NP);
+        final PredictionProvider originalModel = KServeV2GRPCPredictionProvider.forTarget(kServeConfig, new ArrayList<>(request.getData().keySet()), new HashMap<>());
```

Comment:
I am wondering if it would be possible to unify the behavior between this and other explainers by either having the model within the request or set as a service property (see https://github.com/trustyai-explainability/trustyai-explainability/blob/main/explainability-service/src/main/java/org/kie/trustyai/service/endpoints/explainers/ExplainerEndpoint.java). 
I am fine with the former (your proposal here) in the light of making it possible to explain multiple models from the same service. In case we might address this for other explainers in a different issue.

Reply:
@tteofili Good point. I agree it should use that base class. However the TSSaliency is forcing the NP codec, so that might be something we might add the model endpoint builder.
Also, I think the service config host is still referring to the 1:1 model server-TrustyAI service approach, so that might need to change too.

Reply:
- https://github.com/trustyai-explainability/trustyai-explainability/issues/294